pyxem/generators/indexation_generator.py

```diff
@@ -122,6 +123,54 @@ def get_library_FT_dict(template_library, shape, fsize):
     return library_FT_dict
 
 
+class RotationIndexationGenerator:
+    """Generates a template-based indexer that also calculates relative
+    rotation of templates.
+
+    Parameters
+    ----------
+    signal : ElectronDiffraction2D
+        The signal of electron diffraction patterns to be indexed.
+    diffraction_library : DiffractionLibrary
+        The library of simulated diffraction patterns for indexation.
+
+    Notes
+    -----
+    To be used with minimal template libraries whereby the first euler
+    angle is 0. It is this angle that is optimized during indexation.
+    """
+    def __init__(self, signal, diffraction_library):
+        self.signal = signal
+        self.library = diffraction_library
+
+    def correlate(self,
+                  n_largest=5,
+                  method="fast_correlation",
+                  include_phases=None,
+                  optimize_direct_beam=False,
```

Comment:
I really don't think we should do this at this stage (rather than centring the data at an earlier stage), although I have vague memories that you offered a good reason to do it like this?!

Reply:
mainly my own convenience and speed. I wanted to test things on datasets with the fewest number of in-between steps, and I didn't want to modify and resave said data. However, I would also wager many users might appreciate a simplified one-step process.

Reply:
I've come around to this. No change needed